### PR TITLE
CI: optimize git-checkout operations in CI-workflow.

### PR DESCRIPTION
**Pull request by kalessil**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Updates ci-workflow to avoid checkout with the depth of 0, which effectively fetches all branches and commits with a lightweight version of checkout (~25 seconds faster for Building project jobs and linting).

### Testing

Compare the [optimized run](https://github.com/woocommerce/woocommerce/actions/runs/10157370836/job/28087367064?pr=50104) (Build project jobs and Linting) with a [one from a recent PRs](https://github.com/woocommerce/woocommerce/actions/runs/10152693793/job/28074624847): note ~25 seconds difference in checkout times.

Example if triggered linting failure: https://github.com/woocommerce/woocommerce/actions/runs/10164301750/job/28109357670?pr=50104.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -36,9 +36,19 @@ jobs:
       report-jobs: ${{ steps.project-jobs.outputs.report-jobs }}
     steps:
       - uses: 'actions/checkout@v4'
-        name: 'Checkout'
+        name: "Checkout (${{ github.event.pull_request.base.ref || github.base_ref }})"
+        with:
+          ref: ${{ github.event.pull_request.base.ref || github.base_ref }}
+          fetch-depth: 500
```
Comment (kalessil) [inline .github/workflows/ci.yml]: ~3.5 weeks of commits history in trunk.

Comment (adimoldovan) [reply .github/workflows/ci.yml]: Is this a final depth? Why did you choose 500?

**Comment by adimoldovan** [comment]
@kalessil before doing a final test I wonder if it's worth it adding this complexity for the gain we get.
Also, if you think it's worth it, what do you think about dropping 2 steps of actions/checkout with extra logic and simply using a single `run` step that executes the `git` commands we need.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (kalessil) [reply .github/workflows/ci.yml]: Nope, not final one. Picked by combining:
- recommendation for the depth in CI (found GitLab recommendation to be max 1000)
- checking repo stats (~150 merged PRs withing one week) and ensuring the number covers 2-3 weeks of merges

**Comment by kalessil** [comment]
@adimoldovan: introducing single step is a good way to reduce complexity and gives us more flexibility. Let me try it, if it turns simple enough it'll worth the efforts. If still remains complex, then it'll be better to abandon the PR.

**Comment by kalessil** [comment]
No luck here - alternatives are complicating workflow disproportionally to the benefits. Abandoning the PR.

**Pull request closed without merging**